### PR TITLE
feat: CS-783 add workflow for container image build and push to ACR

### PR DESCRIPTION
**Pull request by niteshsakhiya**
## Description
Added workflow for container image build and push to ACR

### CODE REVIEW

File: .github/workflows/build-and-publish-image-acr.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+name: Build and publish container image to ACR
+
+on:
+  push:
```
Comment (bmunguli) [inline .github/workflows/build-and-publish-image-acr.yml:4]: I am a bit unsure about the trigger here if would be push to main for the entry repository. To my understanding we do not intent to develop features and interact with main that much and probably what ever change that we do should go primarily upstream and the main branch can get synched afterwards.

Comment (niteshsakhiya) [reply .github/workflows/build-and-publish-image-acr.yml:4]: Yes. I was also unsure here. But thought of keeping it as it is possible that some of the services might require custom changes. 
I will now remove the on push trigger and later we can add it if needed.

Comment (niteshsakhiya) [reply .github/workflows/build-and-publish-image-acr.yml:4]: Removed the trigger.

File: .github/workflows/build-and-publish-image-acr.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    tags:
+      - 'v*.*.*'
+      - 'v*.*.*-*'
+  workflow_dispatch:
+
+jobs:
+  push-image-to-acr:
+    runs-on: ubuntu-latest
+    environment: acr-shared-env
```
Comment (bmunguli) [inline .github/workflows/build-and-publish-image-acr.yml]: same for the env here

Comment (niteshsakhiya) [reply .github/workflows/build-and-publish-image-acr.yml]: Updated it. Thanks.

**Review by bmunguli — APPROVED** [review]

**Review by typecastcloud — CHANGES_REQUESTED** [review]
@niteshsakhiya 
I would not push changed workflows into ENTRY repository main branch.

If we want to contribute we will have this as code change in PR to upstream.

Alternatives:
- build workflow in separate repo
- branch from release tag, change existing workflow (you can create a branch with this workflow change and merge/cherry-pick into temporary branch)

File: .github/workflows/build-and-publish-image-acr.yml (hunk 1 above)
Comment (bmunguli) [reply .github/workflows/build-and-publish-image-acr.yml:4]: Also another topic here: do we actually want to make changes in the main brach? 
We are considering a way to externalize probably the image building.

**Comment by niteshsakhiya** [comment]
Alternate solution would be used instead of adding pipeline in the main branch.

**Pull request closed without merging**